Add gs_device_rebuild method

borzun opened the pull request:
Add `gs_device_rebuild` method to` graphics.h` to add possibility for manually rebuilding GPU device.

EddyGharbi commented:
1. Could you please resolve the pending merge conflicts?
2. Where is this new API being used?

borzun commented:
> Could you please resolve the pending merge conflicts?
> Where is this new API being used?

1. Yup, will do that;
2. It is used in OSN - https://github.com/stream-labs/obs-studio-node/pull/629

Commit:
Add gs_device_rebuild method

Add gs_device_rebuild method to graphics.h to add possibility for manually rebuilding GPU device.
